### PR TITLE
includeNonPublic and skipDeprecated support #852

### PR DESCRIPTION
**Pull request by pikinier20**
#852 #854

### CODE REVIEW

**Comment by pikinier20** [comment]
Rebased.

**Comment by pikinier20** [comment]
rebased.

**Comment by pikinier20** [comment]
rebased.

File: plugins/base/src/main/kotlin/transformers/documentables/DocumentableFilter.kt [hunk 1]
```diff
@@ -323,5 +339,263 @@ internal class DocumentableVisibilityFilter(val context: DokkaContext) : PreMerg
...
+                        it.extra
+                    )
+                    is DAnnotation -> DAnnotation(
+                        it.name,
+                        it.dri,
+                        it.documentation,
+                        it.expectPresentInSet,
+                        it.sources,
+                        filterFunctions(it.functions).let {
+                            modified = modified || it.first; it.second
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/transformers/documentables/DocumentableFilter.kt:494]: I think there are way to many "it-scopes" open and things should have names at this point 🙄

File: plugins/base/src/main/kotlin/transformers/documentables/DocumentableFilter.kt [hunk 2]
```diff
@@ -5,26 +5,41 @@ import org.jetbrains.dokka.model.*
 import org.jetbrains.dokka.model.DAnnotation
 import org.jetbrains.dokka.model.DEnum
 import org.jetbrains.dokka.model.DFunction
+import org.jetbrains.dokka.model.properties.WithExtraProperties
 import org.jetbrains.dokka.plugability.DokkaContext
 import org.jetbrains.dokka.transformers.documentation.PreMergeDocumentableTransformer
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
-internal class DocumentableVisibilityFilter(val context: DokkaContext) : PreMergeDocumentableTransformer {
+internal class DocumentableFilter(val context: DokkaContext) : PreMergeDocumentableTransformer {
```
Comment (sellmair) [inline plugins/base/src/main/kotlin/transformers/documentables/DocumentableFilter.kt:13]: Wouldn't it be a little bit easier to just implement the visibility filter as well as the deprecated filter as separate transformers in separate files? 🤔

**Pull request merged**